core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -532,7 +532,8 @@ public PersistenceState getPersistenceState(long fileId) throws FileDoesNotExist
       Inode<?> inode = null;
       if (inodePath.fullPathExists()) {
         inode = inodePath.getInode();
-        if (inode.isDirectory() && ((InodeDirectory) inode).isDirectChildrenLoaded()) {
+        if ((inode.isDirectory() && ((InodeDirectory) inode).isDirectChildrenLoaded()) || inode
+            .isFile()) {
```

Comment:
but it also checks if the full path exists, which should evaluate to for a file, correct?